-1

**Failed Tests**:   Build  ClangBuild 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-df0aee/32936/summary.html
**COMMIT**: bfc5432077f75220c6116852f424d4a48d08f0db
**CMSSW**: CMSSW_13_2_X_2023-06-01-1100/el8_amd64_gcc11
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/41836/32936/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Build

I found compilation error when building: 

<pre>>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_13_2_X_2023-06-01-1100/src/GeneratorInterface/Pythia8Interface/plugins/PowhegResHook.cc
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_13_2_X_2023-06-01-1100/src/GeneratorInterface/Pythia8Interface/plugins/Py8toJetInput.cc
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_13_2_X_2023-06-01-1100/src/GeneratorInterface/Pythia8Interface/plugins/Pythia8Hadronizer.cc
In file included from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_13_2_X_2023-06-01-1100/src/GeneratorInterface/Pythia8Interface/plugins/Pythia8Hadronizer.cc:20:
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_13_2_X_2023-06-01-1100/src/GeneratorInterface/Pythia8Interface/plugins/ReweightUserHooks.h: In lambda function:
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_13_2_X_2023-06-01-1100/src/GeneratorInterface/Pythia8Interface/plugins/ReweightUserHooks.h:43:32: error: 'ecms' is not captured
   43 |               pow(1 - 2 * x / (ecms + p[5]), p[1])) *
      |                                ^~~~
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_13_2_X_2023-06-01-1100/src/GeneratorInterface/Pythia8Interface/plugins/ReweightUserHooks.h:41:18: note: the lambda has no capture-default
   41 |     sigma = [this](double x) -> double {
      |                  ^
</pre>

## Clang Build

I found compilation error while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'

```


<pre>>> Local Products Rules ..... started
>> Local Products Rules ..... done
>> Creating project symlinks
>> Entering Package GeneratorInterface/Pythia8Interface
>> Compile sequence completed for CMSSW CMSSW_13_2_X_2023-06-01-1100
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 1
+ eval scram build outputlog '&&' '(python3' /data/cmsbld/jenkins/workspace/ib-run-pr-tests/cms-bot/buildLogAnalyzer.py --logDir /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_13_2_X_2023-06-01-1100/tmp/el8_amd64_gcc11/cache/log/src '||' 'true)'
++ scram build outputlog
>> Entering Package GeneratorInterface/Pythia8Interface
Entering library rule at src/GeneratorInterface/Pythia8Interface/plugins
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_13_2_X_2023-06-01-1100/src/GeneratorInterface/Pythia8Interface/plugins/JetMatchingHook.cc
</pre>

--- a/GeneratorInterface/Pythia8Interface/plugins/ReweightUserHooks.h
+++ b/GeneratorInterface/Pythia8Interface/plugins/ReweightUserHooks.h
@@ -38,7 +38,7 @@ public:
            5.5951279807561e-02,
            3.5e+02};
     const double ecms = (tuneName == "CP5Run3" ? 13600. : 13000.);
-    sigma = [this](double x) -> double {
+    sigma = [this, ecms](double x) -> double {
       return (p[0] * pow(x, p[2] + p[3] * log(0.01 * x) + p[4] * pow(log(0.01 * x), 2)) *
               pow(1 - 2 * x / (ecms + p[5]), p[1])) *
              x;

